### PR TITLE
Refactor reasons mappers and related test

### PR DESCRIPTION
**Pull request by mouhajer-ch**
Related to [BI-8307](https://companieshouse.atlassian.net/browse/BI-8307)
- Refactor all Reason mapper - switched to use setters instead of a constructor
- Refactor AppealMapperTest

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/mapper/AppealMapperTest.java [hunk 1]
```diff
@@ -38,40 +57,27 @@ void shouldReturnNullWhenValueIsNull() {
...
-                ),
-                reason
+                mockCreatedBy,
+                mockPenaltyIdentifier,
+                mockReason
             ));
+
             assertNull(mapped.getId());
             assertNull(mapped.getCreatedAt());
             assertNull(mapped.getCreatedBy());
```
Comment (CH-JonathanBlake) [inline src/test/java/uk/gov/companieshouse/mapper/AppealMapperTest.java:74]: We could probably improve the tests by not using values for these fields.
Or adding a new test where they aren't null.
May be out of scope for a refactor.

File: src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java [hunk 2]
```diff
@@ -17,7 +19,12 @@
 
 @ExtendWith(SpringExtension.class)
 public class OtherReasonMapperTest {
-    private final OtherReasonMapper mapper = new OtherReasonMapper(new AttachmentMapper());
+
+    @Spy
```
Comment (CH-JonathanBlake) [inline src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java:23]: We could mock and use verifies here to make this class more consistent with the others.

File: src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java [hunk 3]
```diff
@@ -38,21 +48,17 @@ void shouldReturnNullWhenValueIsNull() {
...
-            assertEquals(illPerson, mapped.getIllnessReason().getIllPerson());
-            assertEquals(otherPerson, mapped.getIllnessReason().getOtherPerson());
-            assertEquals(illnessStart, mapped.getIllnessReason().getIllnessStartDate());
-            assertEquals(continuedIllness, mapped.getIllnessReason().getContinuedIllness());
-            assertEquals(illnessEnd, mapped.getIllnessReason().getIllnessEndDate());
-            assertEquals(illnessImpactFurtherInformation,
-                mapped.getIllnessReason().getIllnessImpactFurtherInformation());
+
+            assertEquals(illnessReasonEntity, mapped.getIllnessReason());
+            assertEquals(otherReasonEntity, mapped.getOther());
```
Comment (CH-JonathanBlake) [inline src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java:61]: Would be good to see the verifies here

File: src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java (hunk 2 above)
Comment (mouhajer-ch) [reply src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java]: Thanks Jonathan! Improvement done

File: src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java (hunk 3 above)
Comment (mouhajer-ch) [reply src/test/java/uk/gov/companieshouse/mapper/ReasonMapperTest.java:62]: Thanks Jonathan! Improvement done

File: src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java [hunk 4]
```diff
@@ -35,10 +44,16 @@ void shouldReturnNullWhenValueIsNull() {
...
-            OtherReasonEntity mapped = mapper.map(new OtherReason(title, description, Collections.emptyList()));
+            List<Attachment> attachments = new ArrayList<Attachment>();
+            attachments.add(mockAttachment);
+
+            OtherReasonEntity mapped = mapper.map(new OtherReason(title, description, attachments));
+
             assertEquals(title, mapped.getTitle());
             assertEquals(description, mapped.getDescription());
-            assertTrue(mapped.getAttachments().isEmpty());
+            assertEquals(mockAttachment, attachments.get(0));
```
Comment (CH-JonathanBlake) [inline src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java:54]: This is incorrect as you're testing the list you added the mock attachment to, not the attachment list from made by the mapper.
Remember to setup the when for the AttachmentMapper

Comment (mouhajer-ch) [reply src/test/java/uk/gov/companieshouse/mapper/OtherReasonMapperTest.java]: Done!

**Review by CH-JonathanBlake — APPROVED** [review]
